packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -60,53 +59,48 @@
     ],
 }  # type: Dict[FipaMessage.Performative, List[Union[None, FipaMessage.Performative]]]
 
+SUPPLY_DATAMODEL_NAME = "supply"
+DEMAND_DATAMODEL_NAME = "demand"
+
 
 class FipaDialogue(Dialogue):
     """The FIPA dialogue class maintains state of a dialogue and manages it."""
 
-    STARTING_MESSAGE_ID = 1
-    STARTING_TARGET = 0
-
-    class EndState(Enum):
+    class EndState(Dialogue.EndState):
         """This class defines the end states of a dialogue."""
 
         SUCCESSFUL = 0
         DECLINED_CFP = 1
         DECLINED_PROPOSE = 2
         DECLINED_ACCEPT = 3
 
-    class AgentRole(Enum):
-        """This class defines the agent's role in the dialogue."""
+    class AgentRole(Dialogue.Role):
+        """This class defines the agent's role in a fipa dialogue."""
 
         SELLER = "seller"
         BUYER = "buyer"
 
     def __init__(
-        self, dialogue_label: DialogueLabel, is_seller: bool, **kwargs
+        self, dialogue_label: DialogueLabel, role: AgentRole, **kwargs
     ) -> None:
```

Comment:
`**kwargs` can be dropped, we do not allow for them here.

Reply:
Remove the whole method

Done